src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -391,6 +405,45 @@ private BulkResponse execute() throws Exception {
         }
       }
     }
+
+    private void handleMalformedDoc(BulkResponse bulkRsp) {
+      // if the elasticsearch request failed because of a malformed document,
+      // the behavior is configurable.
+      switch (behaviorOnMalformedDoc) {
+        case WARN:
+          log.warn("Encountered an illegal document error when executing batch {} of {}"
+                  + " records. Error was {}. Will not index record.",
+              batchId, batch.size(), bulkRsp.getErrorInfo());
+          return;
+        case IGNORE:
+          if (log.isDebugEnabled()) {
```

Comment:
You probably can skip this check. As elsewhere in the code, the log level is not checked when `batch.size()` is called, because the collection is an array list. Since calling `getErrorInfo` is also negligible we can skip the extra code here.